### PR TITLE
Mo' colors

### PR DESCRIPTION
**Pull request by C0DK**
Changes:
- Stream is colored now
- Stream is formatted
- Stream has it's own formatting option
- Colors are now a style, and is a struct
- Color struct has a pretty cool functional interface
- colored mentions and PMs
- Every message uses the same function (it's dry!!)
- Colorize errors!
- Create function for visualizing errors
- colorized some of the command output!
- Color is stored in a Style
- Create a Text struct that can use to stylize strings "easily"
- Text can be used to build strings
- color highlighting on code
- added tml config support
- added different color for mention url
- Added sprintf to use formatting with PrintFeed and PrintError

Known Bugs: (added as todos whereever)
- Cannot use multiple formatting at the same time (*bold _italic_*
doesn't work
- sprintf is pretty shit
- background doesn't cover as a `block` in codeblocks
- not possible to escape sprintf thing

### CODE REVIEW

File: colors.go [hunk 1]
```diff
@@ -3,41 +3,142 @@ package main
...
-func colorReplaceMentionMe(msg string, offColor string) string {
-	//var coloredOwnName = colorText(k.Username, mentionColor, offColor)
-	//return strings.Replace(msg, k.Username, coloredOwnName, -1)
-	return colorRegex(msg, "(@?"+k.Username+")", mentionColor, offColor)
+// Appends the other text at the end, but retains same style
+func (txt Text) append(other Text) Text {
+	txt.message = txt.message + other.stringFollowedByStyle(txt.style)
+	return txt
+}
+func (txt Text) add(other string) Text {
```
Comment (C0DK) [inline colors.go]: change to "appendString"

File: cmdDelete.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ func cmdDelete(cmd []string) {
 	chat := k.NewChat(channel)
 	_, err := chat.Delete(messageID)
 	if err != nil {
-		printToView("Feed", fmt.Sprintf("There was an error deleting your message."))
+		printError(fmt.Sprintf("There was an error deleting your message."))
```
Comment (C0DK) [inline cmdDelete.go]: remove sprintf

File: colors.go [hunk 3]
```diff
@@ -3,41 +3,142 @@ package main
...
+	strikethrough   bool
+}
+
+var basicStyle = Style{normal, normal, false, false, false, false}
+
+func styleFromString(input string) Style {
+	return basicStyle
+}
+
+func (style Style) withForeground(f color) Style {
```
Comment (C0DK) [inline colors.go]: change to `s` instead of `style`

**Comment by Rudi9719** [comment]
Please make sure you're using the `mage.go` and `.travis.yml` from dev, your PR failed to build, looks like it didn't update the Keybase fwrk?

File: cmdDelete.go (hunk 2 above)
Comment (Rudi9719) [reply cmdDelete.go]: Please do

**Review by Rudi9719 — COMMENTED** [review]
Please check your `.travis.yml` file, `mage.go` and `cmdDelete.go` - you commented about removing sprintf but didn't?

**Review by Rudi9719 — CHANGES_REQUESTED** [review]
Please make the changes commented above :+1: Forgot to mark "Request changes" before

**Comment by C0DK** [comment]
I'l look at the travis stuff soon. not sure what went wrong there :'( but i'll figure it out soon.

But afaik i've fixed all comments and bugs (not mentioned in the commit message :P )

**Comment by C0DK** [comment]
it passed now!

**Review by Rudi9719 — APPROVED** [review]

**Pull request merged**